mne_bids/utils.py

```diff
@@ -264,6 +264,30 @@ def make_dataset_description(path, name=None, data_license=None,
     _write_json(description, fname, verbose=verbose)
 
 
+def age_on_date(bday, exp_date):
+    """
+    Determine the age of someone from their birthday and the date of the
+    experiment.
+
+    Parameters
+    ----------
+    bdat : instance of datetime.datetime
+        The birthday of the participant.
+    exp_date : instance of datetime.datetime
+        The date the experiment was performed on.
+    """
+    if exp_date > bday:
+        if exp_date.month > bday.month:
+            return exp_date.year - bday.year
+        elif exp_date.month == bday.month:
+            if exp_date.day >= bday.day:
+                return exp_date.year - bday.year
+        return exp_date.year - bday.year - 1
+    else:
+        raise ValueError("The experimentation data must be after the birth "
+                         "date")
```

Comment:
always raise the `ValueError` first so you can avoid this extra level of nesting for `if-else`